modules/stochastic_tools/include/utils/GaussianProcessHandler.h

```diff
@@ -52,11 +52,15 @@ class GaussianProcessHandler
    * @param tao_options Additional options if TAO is used for parameter optimization.
    * @param show_tao Switch to show details of TAO optimization.
    */
-  void setupCovarianceMatrix(const RealEigenMatrix & training_params,
-                             const RealEigenMatrix & training_data,
-                             MooseEnum opt_type,
-                             std::string tao_options = "",
-                             bool show_tao = false);
+  void
+  setupCovarianceMatrix(RealEigenMatrix training_params, // const RealEigenMatrix & training_params
```

Comment:
```suggestion
  setupCovarianceMatrix(RealEigenMatrix training_params,
```

Reply:
Also it would be nice if we could keep the reference here. You could create another matrix when you shuffle (as you already do with `input` and `output`) and just operate on those (pass those to `getAdamLoss` and `Gradient` functions) in the batches instead of manipulating the member variables. I think it would be optimal to keep the original data intact.